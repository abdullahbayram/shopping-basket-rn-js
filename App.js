import './wdyr'; // <--- first import

import { StatusBar } from 'expo-status-bar';
import { StyleSheet } from 'react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { Provider } from 'react-redux';
import { MD3LightTheme, PaperProvider } from 'react-native-paper';
import * as React from 'react';
import store from './src/redux/store';
import Navigator from './src/Navigator';

const theme = {
  ...MD3LightTheme, // or MD3DarkTheme
  roundness: 2,
  colors: {
    ...MD3LightTheme.colors,
    primary: '#3498db',
    secondary: '#f1c40f',
    tertiary: '#a1b2c3',
  },
};

const App = () => {
  return (
    <SafeAreaProvider testID="SafeAreaProvider" style={styles.container}>
      <Provider store={store}>
        <PaperProvider theme={theme}>
          {/* eslint-disable-next-line react/style-prop-object */}
          <StatusBar style="auto" />
          <Navigator />
        </PaperProvider>
      </Provider>
    </SafeAreaProvider>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});

export default App;
