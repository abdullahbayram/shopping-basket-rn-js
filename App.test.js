import { render, screen } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import App from './App';

const mockStore = configureStore([]);
const store = mockStore({});

describe('<App />', () => {
  test('Text renders correctly on App', () => {
    const { toJSON } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    screen.getByTestId('SafeAreaProvider');
    expect(toJSON()).toMatchSnapshot();
  });
});
