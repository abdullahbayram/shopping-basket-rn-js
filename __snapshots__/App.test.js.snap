// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Text renders correctly on App 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
    ]
  }
  testID="SafeAreaProvider"
/>
`;
